lib/experimental/class-wp-webfonts.php

```diff
@@ -14,238 +14,332 @@
-		return $this->providers;
+	public function remove_font_family( $font_family_handle ) {
+		if ( ! isset( $this->registered[ $font_family_handle ] ) ) {
+			return;
+		}
+
+		$variations = $this->registered[ $font_family_handle ]->deps;
+
+		foreach ( $variations as $variation ) {
+			$this->remove( $variation );
+		}
+
+		$this->remove( $font_family_handle );
 	}
 
 	/**
-	 * Register a webfont.
+	 * Registers a variation to the given font family.
 	 *
-	 * @since 6.0.0
+	 * @since X.X.X
 	 *
-	 * @param array $webfont Webfont to be registered.
-	 * @return string|false The font family slug if successfully registered, else false.
+	 * @param string $font_family_handle The font family's handle for this variation.
+	 * @param array  $variation          An array of variation properties to add.
+	 * @param string $variation_handle   Optional. The variation's handle. When none is provided, the
+	 *                                   handle will be dynamically generated.
+	 *                                   Default empty string.
+	 * @return string|null Variation handle on success. Else null.
 	 */
-	public function register_webfont( array $webfont ) {
-		$webfont = $this->validate_webfont( $webfont );
+	public function add_variation( $font_family_handle, array $variation, $variation_handle = '' ) {
+		if ( ! WP_Webfonts_Utils::is_defined( $font_family_handle ) ) {
+			trigger_error( 'Font family handle must be a non-empty string.' );
+			return null;
+		}
 
-		// If not valid, bail out.
-		if ( ! $webfont ) {
-			return false;
+		if ( '' !== $variation_handle && ! WP_Webfonts_Utils::is_defined( $variation_handle ) ) {
+			trigger_error( 'Variant handle must be a non-empty string.' );
+			return null;
+		}
+
+		// Register the font family when it does not yet exist.
+		if ( ! isset( $this->registered[ $font_family_handle ] ) ) {
+			if ( ! $this->add_font_family( $font_family_handle ) ) {
+				return null;
+			}
 		}
 
-		$slug = $this->get_font_slug( $webfont );
+		$variation = $this->validate_variation( $font_family_handle, $variation );
 
-		// Initialize a new font-family collection.
-		if ( ! isset( $this->registered_webfonts[ $slug ] ) ) {
-			$this->registered_webfonts[ $slug ] = array();
+		// Variation validation failed.
+		if ( ! $variation ) {
+			return null;
 		}
 
-		$this->registered_webfonts[ $slug ][] = $webfont;
-		return $slug;
+		if ( '' === $variation_handle ) {
+			$variation_handle = WP_Webfonts_Utils::convert_variation_into_handle( $font_family_handle, $variation );
+			if ( is_null( $variation_handle ) ) {
+				return null;
+			}
+		}
+
+		// Bail out if the variant is already registered.
+		if ( $this->is_variation_registered( $font_family_handle, $variation_handle ) ) {
+			return $variation_handle;
+		}
+
+		if ( array_key_exists( 'src', $variation ) ) {
+			$result = $this->add( $variation_handle, $variation['src'] );
+		} else {
+			$result = $this->add( $variation_handle, false );
+		}
```

Comment:
Just a nitpick and not important... I personally just avoid using `else` when possible, I find that in most cases it's unnecessary
```suggestion
		$variation_src = array_key_exists( 'src', $variation ) ? $variation['src'] : false;
		$result        = $this->add( $variation_handle, $variation_src );
```

Reply:
The ternary design does not eliminate the  `else` (i.e. `:` is an `else`). Rather, it's a more compressed version of `if/else` code structure. I'm fine with compressing it as in this case it does eliminate redundancy while providing more information about that 2nd argument being passed to the `add()`.